common/scala/src/main/scala/whisk/core/containerpool/Container.scala

```diff
@@ -70,7 +70,10 @@ trait Container {
   protected var httpConnection: Option[ContainerClient] = None
 
   /** Stops the container from consuming CPU cycles. */
-  def suspend()(implicit transid: TransactionId): Future[Unit]
+  def suspend()(implicit transid: TransactionId): Future[Unit] = {
+    //close connection and reset the connection pool
+    closeConnections().andThen { case _ => httpConnection = None }
```

Comment:
I think this should set `httpConnection = None` before running `closeConnections`. Otherwise you might get connections running on a closing client when you unpause directly.